app/service/file_svc.py

```diff
@@ -96,6 +115,22 @@ def __init__(self, plugins, exfil_dir):
         """
         self.special_payloads[name] = func
 
+    async def find_payloads(self):
+        """
+        Identify the full gamut of available payloads (filtering out adversary mode)
+        :return: list of available payloads
+        """
+        listing = []
+        for plugin in self.plugins:
+            for root, _, files in os.walk('plugins/%s' % plugin):
+                if root.endswith('payloads') and 'adversary' not in root:
+                    listing.extend(files)
+            for _, _, files in os.walk('data/payloads'):
+                for file in files:
+                    if not file.endswith('.gitkeep'):
+                        listing.append(file)
```

Comment:
Seems like you don't need this to be part of the parent loop?

Reply:
Good catch, fixed.